notebook/static/notebook/js/notebook.js

```diff
@@ -1802,7 +1802,7 @@ define(function (require) {
         function restart_and_resolve () {
             that.kernel.restart(function () {
                 // resolve when the kernel is *ready* not just started
-                that.events.one('kernel_ready.Kernel', resolve_promise);
+                that.events.trigger('kernel_ready.Kernel', resolve_promise);
```

Comment:
Same here - this is *adding* a trigger for kernel_ready and unregistering a callback.